server/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -129,6 +130,16 @@ private EqMatches extractMatches(Collection<ColumnIdent> columns,
             }
         }
 
+        // the result of equality extractor is invalid if the unknowns contain `<ref> IS NULL`,
+        // ex) `a = 1 or a is null` -> a = 1 is meaningless
+        for (var unknown: context.unknowns) {
+            if (unknown instanceof Function function
+                && function.signature().equals(IsNullPredicate.SIGNATURE)
+                && context.comparisons.containsKey(((Reference)function.arguments().get(0)).column())) {
+                return EqMatches.NONE;
+            }
+        }
+
```

Comment:
If handled this way, it becomes difficult to identify either `a=1 <op> a is null` or `<op>(a=1 or a is null)` or both `<op>`.

Reply:
Not taking this approach.